app/models/type_de_champ.rb

```diff
@@ -24,6 +26,9 @@ class TypeDeChamp < ActiveRecord::Base
 
   belongs_to :procedure
 
+  scope :public_only, -> { where.not(type: 'TypeDeChampPrivate').or(where(private: [false, nil])) }
```

Comment:
`where.not(type: 'TypeDeChampPrivate')`
C'est pour gérer les données avant la migration ?

Reply:
oui